extensions/notebook/src/book/bookPinManager.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as constants from './../common/constants';
+import { BookTreeItem } from './bookTreeItem';
+import { getPinnedNotebooks } from '../common/utils';
+
+export interface IBookPinManager {
+	pinNotebook(notebook: BookTreeItem): boolean;
+	unpinNotebook(notebook: BookTreeItem): boolean;
+}
+
+enum PinBookOperation {
+	Pin,
+	Unpin
+}
+
+export class BookPinManager implements IBookPinManager {
+
+	constructor() {
+		this.setPinnedSectionContext();
+	}
+
+	setPinnedSectionContext(): void {
+		if (getPinnedNotebooks().length > 0) {
+			vscode.commands.executeCommand(constants.BuiltInCommands.SetContext, constants.showPinnedBooksContextKey, true);
+		} else {
+			vscode.commands.executeCommand(constants.BuiltInCommands.SetContext, constants.showPinnedBooksContextKey, false);
+		}
+	}
+
+	isNotebookPinned(notebookPath: string): boolean {
+		if (getPinnedNotebooks().findIndex(x => x === notebookPath) > -1) {
+			return true;
+		}
+		return false;
+	}
+
+	pinNotebook(notebook: BookTreeItem): boolean {
+		return this.isNotebookPinned(notebook.book.contentPath) ? false : this.updatePinnedBooks(notebook, PinBookOperation.Pin);
```

Comment:
So this returns false if the notebook isn't actually pinned but unpinNotebook always returns updatePinnedBooks directly? It looks like the logic for updatePinnedBooks will return true/false as appropriate so you should be able to get rid of that check here right?